tests/worker/topics/test_rabbitmq_topic.py

```diff
@@ -212,3 +212,30 @@ async def test_rabbitmq_topic_expiring_message(
         assert await unwrap(await alib.anext(topic_iter)) == message
 
     await topic.close()
+
+
+@pytest.mark.asyncio
+async def test_retry(
+    rabbitmq_topic_maker: t.Callable[..., Awaitable[RabbitMQTopic]]
+) -> None:
+    topic = await rabbitmq_topic_maker(
+        RabbitMQTopic, serializer=RabbitMQSerializer.PICKLE, max_retry=2
+    )
+
+    messages = [
+        TopicMessage(id=MessageId("0"), args={"n": b"1", "time": utcnow()}),
+    ]
+
+    for message in messages:
+        await topic.publish(message, wait=True)
+
+    async with alib.scoped_iter(topic.run()) as topic_iter:
+        context = await alib.anext(topic_iter)
+        with pytest.raises(ValueError):
+            async with context as message:
+                raise ValueError("Exception")
+        context = await alib.anext(topic_iter)
+        async with context as message:
+            assert message.id == "0"
```

Comment:
you should test that we retry up to `max_retry` tries?

Reply:
I tried to figure a way, but the only way I found is ensure we are waiting infinitely for a message at the end of the test..